bokehjs/src/lib/models/math_text/math_text.ts

```diff
@@ -162,12 +193,70 @@ export class MathTextView extends View {
     }
   }
 
-  private get_dimensions(): Size {
+  _size(): Size {
     return this.has_image_loaded
       ? this.get_image_dimensions()
       : this.get_text_dimensions()
   }
 
+  bbox(): BBox {
+    const {p0, p1, p2, p3} = this.rect()
+
+    const left = Math.min(p0.x, p1.x, p2.x, p3.x)
+    const top = Math.min(p0.y, p1.y, p2.y, p3.y)
+    const right = Math.max(p0.x, p1.x, p2.x, p3.x)
+    const bottom = Math.max(p0.y, p1.y, p2.y, p3.y)
+
+    return new BBox({left, right, top, bottom})
+  }
+
+  rect(): Rect {
+    const rect = this._rect()
+    const {angle} = this
+    if (!angle)
+      return rect
+    else {
+      const {sx, sy} = this.position
+      const tr = new AffineTransform()
+      tr.translate(sx, sy)
+      tr.rotate(angle)
+      tr.translate(-sx, -sy)
+      return tr.apply_rect(rect)
+    }
+  }
+
+  paint_rect(ctx: Context2d): void {
+    const {p0, p1, p2, p3} = this.rect()
+    ctx.save()
+    ctx.strokeStyle = "red"
+    ctx.lineWidth = 1
+    ctx.beginPath()
+    const {round} = Math
+    ctx.moveTo(round(p0.x), round(p0.y))
+    ctx.lineTo(round(p1.x), round(p1.y))
+    ctx.lineTo(round(p2.x), round(p2.y))
+    ctx.lineTo(round(p3.x), round(p3.y))
+    ctx.closePath()
+    ctx.stroke()
+    ctx.restore()
+  }
+
+  paint_bbox(ctx: Context2d): void {
+    const {x, y, width, height} = this.bbox()
+    ctx.save()
+    ctx.strokeStyle = "blue"
+    ctx.lineWidth = 1
+    ctx.beginPath()
+    const {round} = Math
+    ctx.moveTo(round(x), round(y))
+    ctx.lineTo(round(x), round(y + height))
+    ctx.lineTo(round(x + width), round(y + height))
+    ctx.lineTo(round(x + width), round(y))
+    ctx.closePath()
+    ctx.stroke()
+    ctx.restore()
+  }
```

Comment:
are these functions just for diagnostic/debugging purposes?

Reply:
I needed to add those for compatibility with GraphicsBox, since tick labels are an array of GraphicBox. It's exactly use is not clear to me. Maybe @mattpap can explain a bit about it

Reply:
> are these functions just for diagnostic/debugging purposes?

Yes. Ultimately those will be inherited from `GraphicsBox` when the class structure is resolved.